src/transformers/generation/utils.py

```diff
@@ -301,9 +301,9 @@ class BeamSearchDecoderOnlyOutput(ModelOutput):
             of log probabilities of tokens conditioned on log softmax of previously generated tokens in this beam.
             Tuple of `torch.FloatTensor` with up to `max_new_tokens` elements (one element for each generated token),
             with each tensor of shape `(batch_size*num_beams*num_return_sequences, config.vocab_size)`.
-        beam_indices (`tuple(tuple(torch.LongTensor))`, *optional*, returned when `output_scores=True` is passed or when `config.output_scores=True`):
+        beam_indices (`torch.LongTensor`, *optional*, returned when `output_scores=True` is passed or when `config.output_scores=True`):
             Beam indices of generated token id at each generation step. `torch.LongTensor` of shape
-            `(batch_size*num_return_sequences, input_ids.shape[-1])`.
+            `(batch_size*num_return_sequences, sequence_length)`.
```

Comment:
Guess it's a bit difficult for the user to compute what `sequence_length` is, but `input_ids.shape[-1]` is also not great. I wonder whether it could make sense to add a small doc that states how `sequence_length` is computed depending on the different use cases? Given how popular `generate` is, might be worth it ?

Reply:
Yes, a user asked what `sequence_length` means two days ago! I'll add a TODO to try to improve these docs